pkg/controllers/user/alert/watcher/node.go

```diff
@@ -90,43 +91,63 @@ func (w *NodeWatcher) watchRule() error {
 			}
 			for _, node := range nodes {
 				machine := getMachineByNodeName(machines, node.Name)
-				w.checkNodeCondition(node, alert, machine)
+				w.checkNodeCondition(alert, machine)
 			}
 		}
 	}
 	return nil
 }
 
-func getMachineByNodeName(machines []*v3.Node, nodeName string) *v3.Node {
+func getMachineByID(machines []*v3.Node, id string) *v3.Node {
 	for _, m := range machines {
-		if m.Status.NodeName == nodeName {
+		if m.Name == id {
 			return m
 		}
 	}
+	return nil
+}
 
+func getMachineByNodeName(machines []*v3.Node, nodeName string) *v3.Node {
+	for _, m := range machines {
+		if getNodeName(m) == nodeName {
+			return m
+		}
+	}
 	return nil
+}
 
+func getNodeName(machine *v3.Node) string {
```

Comment:
this code looks familiar  - we use something similar in provisioner. Lets think about putting it to some kind of a common file, and use from both controllers.

Reply:
ok